deployment/helm/resource-management-policies/balloons/templates/configmap_enable_nri.yaml

```diff
@@ -0,0 +1,36 @@
+{{- if .Values.nri.enable }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: patch-containerd-config
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "balloons-plugin.labels" . | nindent 4 }}
+data:
+  patch_nri.sh: |
+   #!/bin/bash
+
+   config_file="/home/config.toml"
+   plugin="[plugins.\"io.containerd.nri.v1.nri\"]"
+   switch="disable = false"
+   
+   cp /etc/containerd/config.toml "$config_file"
+
+   # Check if the nri entry exists in the config file
+   if grep -q "$plugin" "$config_file"; then
+     # Check if the nri is already enabled
+     if grep -q "disable = false" "$config_file"; then
+       echo "NRI is already enabled..."
+     else
+       # Enable the NRI plugin
+       sed -i "s/disable = true/$switch/" "$config_file"
+       cp -f "$config_file" /etc/containerd/config.toml
+       echo "Restarting containerd...."
+       nsenter -t 1 -m bash
+       systemctl restart containerd
```

Comment:
Does this really work as intended ?

I wonder about that `nsenter` usage. AFAICT, here it starts `bash` in the (mount) namespace of init/pid 1. Then once that bash exits, the original patch_nri.sh script continues in its original namespaces and runs `systemctl restart containerd`. So what does that `nsenter` accomplish ?

Reply:
> Does this really work as intended ?
> 
Yes
> I wonder about that `nsenter` usage. AFAICT, here it starts `bash` in the (mount) namespace of init/pid 1. Then once that bash exits, the original patch_nri.sh script continues in its original namespaces and runs `systemctl restart containerd`. So what does that `nsenter` accomplish ?

We don't exit the bash of the pid1 until we run `systemctl restart containerd`. We enter the pid1 in `nsenter -t 1 -m bash` and we don't exit but instead we just kill the init container as a whole. So, `nsenter` is entering the mount namespace of the init process (PID 1) and launching a bash within that namespace.  For having the access to the host pid, I added  `hostPID: true.`

Reply:
I think that nsenter-started bash exits immediately (when running without a controlling tty) and `systemctl restart containerd` is run in the same namespace as the main script.

Anyway, I wonder what `nsenter -t 1 -m` would anyway change from the namespaces point of view. Namespaces are inherited across forks. If a process is not actively changing its namespaces, it should be running in the same namespace as its parent process, so transitively as PID 1/its init process.

Reply:
Yeah, the nsenter bash looks like nop to me, too

Reply:
It seems that relying on `nsenter -t 1 -m` wasn't sufficient to execute the systemctl... command from within a script, even though it functions well when manually executed in the current terminal session. I have now used `nsenter -t 1 -m systemctl restart containerd` instead, thanks to the help of @klihub .

Here is the short POC:

[![POC](https://gist.github.com/fmuyassarov/bbf2bc8ab55927d1507d78b0190a87b4/raw/3263719e7c198f21e022c8a8b721bf9193927e57/asci.png)](https://asciinema.org/a/sTsXlRdPA8mgkyxB9KNVzyggt)